doc/source/development/extending.rst

```diff
@@ -208,13 +208,201 @@ will
+how your ExtensionArray should interact with numpy's dispatch logic
+in order to achieve its goal, since there are several alternative ways
+of achieving similar results.
+
+However, the details of numpy's dispatch logic are not entirely simple, and
+there nuances which you should be aware of. For that reason, and in order to
+make it easier to create new pandas extensions, we will now cover some of
+possible approaches of dealing with numpy.
+
+The first alternative, and the simplest, is to simply provide an `__array__`
+method for your ExtensionArray. This is a standard numpy function documented
+here (TBD), which must return a numpy array equivalent of your ExtensionArray.
+This will usually be an array whose dtype is `object` and whose values are
+instances of some class which your ExtensionArray wraps into an array. For
+example, the pandas tests include an ExtensionArray example called
+`DecimalArray`, if it used this method, its `__array__` method would return an
+ndarray of `decimal.Decimal` objects.
+
+Implementing `__array__`  is easy, but it usually isn't satisfactory because
+it means most Series operations will return a Series of object dtype, instead
+of maintaining your ExtensionArray's dtype.
+
+The second approach is more involved, but it does a proper job of maintaining
+the ExtensionArray's dtype through operations. It requires a detailed
+understanding of how numpy functions operate on non ndarray objects.
+
+Just as pandas handles some operation via :meth:`ExtensionArray._reduce`
+and others by delegating to numpy, numpy makes a distinction between
+between two types of operations: ufuncs (such as `np.floor`, `np.ceil`,
+and `np.abs`), and non-ufuncs (for example `np.round`, and `np.repeat`).
+
+We will deal with ufuncs first. You can find a list of numpy's ufuncs here
+(TBD). In order to support numpy ufuncs, a convenient approach is to implement
+numpy's `__array_ufunc__` interface, specified in
+[NEP13](https://www.numpy.org/neps/nep-0013-ufunc-overrides.html). In brief,
+if your ExtensionArray implements a compliant `__array_ufunc__` interface,
+when a numpy ufunc such as `np.floor` is invoked on your array, its
+implementation of `__array_ufunc__`  will bec called first and given the
+opportunity to compute the function. The return value needn't be a numpy
+ndarray (though it can be). In general, you want the return value to be an
+instance of your ExtensionArray. In some cases, your implementation can
+calculate the result itself (see for example TBD), or, if your ExtensionArray
+already has a numeric ndarray backing it, your implementation will itself
+invoke the numpy ufunc itself on it (see for example TBD). In either case,
+after computing the values, you will usually want to wrap the result as a new
+ExtensionArray instance and return it to the caller. Pandas will automatically
+use that Array as the backing ExtensionArray for a new Series object.
+
+.. note::
```

Comment:
Can remove this. It's documented in the NEP.

Reply:
If at all possible, I hoped this doc would spare EA developers the need to the NEP too carefully.
But I guess it is redundant. I'll remove it.

Reply:
This is for rather old numpy versions, which pandas will not even support anymore starting from 0.25.0 (IIRC). So that's another reason it can be removed.